### ISSUE TITLE
All zod model properties wrapped in z.ZodOptional

### ISSUE THREAD

**Issue by tlancina** [zenstackhq/zenstack#898]
Hello! I guess it's more of a question: is there a reason for all model properties to be optional? I just upgraded to 1.5.0 from 1.2.0 and it broke a few types. I filed it as a bug since already optional fields are marked as optional again.

1.2.0:
```typescript
export declare const AccountSchema: z.ZodObject<{
    id: z.ZodString;
    type: z.ZodString;
    provider: z.ZodString;
    providerAccountId: z.ZodString;
    refresh_token: z.ZodOptional<z.ZodNullable<z.ZodString>>;
    access_token: z.ZodOptional<z.ZodNullable<z.ZodString>>;
    expires_at: z.ZodOptional<z.ZodNullable<z.ZodNumber>>;
    token_type: z.ZodOptional<z.ZodNullable<z.ZodString>>;
    scope: z.ZodOptional<z.ZodNullable<z.ZodString>>;
    id_token: z.ZodOptional<z.ZodNullable<z.ZodString>>;
    session_state: z.ZodOptional<z.ZodNullable<z.ZodString>>;
}

```
1.5.0:
```typescript
export declare const AccountSchema: z.ZodObject<{
    type: z.ZodOptional<z.ZodString>;
    id: z.ZodOptional<z.ZodString>;
    provider: z.ZodOptional<z.ZodString>;
    providerAccountId: z.ZodOptional<z.ZodString>;
    refresh_token: z.ZodOptional<z.ZodOptional<z.ZodNullable<z.ZodString>>>;
    access_token: z.ZodOptional<z.ZodOptional<z.ZodNullable<z.ZodString>>>;
    expires_at: z.ZodOptional<z.ZodOptional<z.ZodNullable<z.ZodNumber>>>;
    token_type: z.ZodOptional<z.ZodOptional<z.ZodNullable<z.ZodString>>>;
    scope: z.ZodOptional<z.ZodOptional<z.ZodNullable<z.ZodString>>>;
    id_token: z.ZodOptional<z.ZodOptional<z.ZodNullable<z.ZodString>>>;
    session_state: z.ZodOptional<z.ZodOptional<z.ZodNullable<z.ZodString>>>;
    user: z.ZodOptional<z.ZodRecord<z.ZodString, z.ZodUnknown>>;
    userId: z.ZodOptional<z.ZodString>;
}
```

**Environment (please complete the following information):**

-   ZenStack version: 1.5.0

I haven't had a chance to dig into why (or bisect what version exactly this happened at), just posting in case it's immediately clear. Will dig in more when I get a chance.

**Comment by ymc9** [comment]
Hi @tlancina ,

Thanks for bringing this up. Let me provide some clarifications about the generated zod schemas.

The main schemas (as you showed above `AccountSchema`) are meant to represent the entity fetched with a prisma `findXXX` call. Since with Prisma you can load entities partially using "select" and "include", I feel it makes sense to mark all fields optional. There are two variant zod schemas generated:

- `AccountCreateSchema`
    All non-optional ZModel fields that don't have default values are required. It's meant for validating a create payload.

- `AccountUpdateSchema`

    All fields are optional. The difference with the main schema is that for relations, this variant only contains foreign key fields, but not the relation model fields. It's meant for validating update payload.

It seems that when calling `z.partial` on a zod object, "nullish" fields are wrapped with multiple layers of `ZodOptional` typing. However, I didn't notice how this breaks things either at compile or at runtime.

Could you let me know about more specifics on the way it's broken to you? Thanks!

**Comment by tlancina** [comment]
Hey thanks for the explanation. My gut reaction is that if Zod is for validating data going in to Prisma, then something like `partial` should be opt in, but I don't feel particularly strongly about it (I can also equally call `.required`).

At the end of the day this is a nitpick - I don't have a good use case and wasn't even using it to validate input, I was using it to access Enum values as literals for the UI using `.shape` 😅 (and then had to `.unwrap` them).

**Comment by ymc9** [comment]
Fixed in v2.0.0-beta.1
